src/LanguageServer/Test/RdtTests.cs

```diff
@@ -63,5 +70,91 @@ public void TestInitialize()
             rdt.GetDocument(uri1).Should().BeNull();
             ds.Diagnostics.TryGetValue(uri1, out _).Should().BeFalse();
         }
+
+        [TestMethod, Priority(0)]
+        public async Task LockCount() {
+            var uri1 = TestData.GetDefaultModuleUri();
+            await CreateServicesAsync(PythonVersions.LatestAvailable3X, uri1.AbsolutePath);
+            var rdt = Services.GetService<IRunningDocumentTable>();
+
+            rdt.OpenDocument(uri1, "from LockCount1 import *");
+            var doc = rdt.GetDocument(uri1);
+            await doc.GetAstAsync(CancellationToken.None);
+            await Services.GetService<IPythonAnalyzer>().WaitForCompleteAnalysisAsync();
+
+            var docLc1 = rdt.First(d => d.Name.Contains("LockCount1"));
+            var docLc2 = rdt.First(d => d.Name.Contains("LockCount2"));
+            var docLc3 = rdt.First(d => d.Name.Contains("LockCount3"));
+
+            VerifyLockCount(rdt, docLc1.Uri, 1);
+            VerifyLockCount(rdt, docLc2.Uri, 1);
+            VerifyLockCount(rdt, docLc3.Uri, 1);
+
+            rdt.OpenDocument(docLc1.Uri, null);
+            VerifyLockCount(rdt, docLc1.Uri, 2);
+
+            rdt.OpenDocument(docLc3.Uri, null);
+            VerifyLockCount(rdt, docLc3.Uri, 2);
+
+            rdt.CloseDocument(docLc1.Uri);
+            VerifyLockCount(rdt, docLc1.Uri, 1);
+
+            rdt.CloseDocument(docLc3.Uri);
+            VerifyLockCount(rdt, docLc3.Uri, 1);
+        }
+
+        [TestMethod, Priority(0)]
+        public async Task OpenCloseAnalysis() {
+            var uri = TestData.GetDefaultModuleUri();
+            await CreateServicesAsync(PythonVersions.LatestAvailable3X, uri.AbsolutePath);
+            var rdt = Services.GetService<IRunningDocumentTable>();
+
+            rdt.OpenDocument(uri, "from LockCount1 import *");
+            var doc = rdt.GetDocument(uri);
+            await doc.GetAstAsync(CancellationToken.None);
+            await Services.GetService<IPythonAnalyzer>().WaitForCompleteAnalysisAsync();
+
+            var docLc1 = rdt.First(d => d.Name.Contains("LockCount1"));
```

Comment:
`rdt.GetDocument`?

Reply:
Yeah, it was just so they all look the same since I know uri for for first but not others